nessie/src/main/java/org/apache/iceberg/nessie/NessieUtil.java

```diff
@@ -111,23 +111,31 @@ public static TableMetadata updateTableMetadataWithNessieSpecificProperties(
     // Update the TableMetadata with the Content of NessieTableState.
     Map<String, String> newProperties = Maps.newHashMap(tableMetadata.properties());
     newProperties.put(NessieTableOperations.NESSIE_COMMIT_ID_PROPERTY, reference.getHash());
+
     // To prevent accidental deletion of files that are still referenced by other branches/tags,
```

Comment:
Nit: wonder whether all the GC related warning code should better go into a separate method (it's quite some lines of code/comments)

Reply:
good point - moved